### ISSUE TITLE
[FORMATTING] Inconsistent placement of offset index in select - bigquery

### ISSUE THREAD

**Issue by m-ahmadkhan** [sql-formatter-org/sql-formatter#243]
**Input data**

I'm trying to format the following BigQuery query:
```sql
SELECT CAST(SPLIT(
        key1.path, ',')[OFFSET(1)] AS INT64) AS col1,
    CAST(SPLIT(key2.path, ',')[OFFSET(1)] AS INT64) AS col2,
    key3_looooooooooooooong_name.id AS col3
FROM `my_dataset.my_table`;
```

With the formatting options:
```
{
  "language": "bigquery",
  "tabWidth": 2,
  "useTabs": false,
  "keywordCase": "upper",
  "indentStyle": "standard",
  "multilineLists": 4,
  "logicalOperatorNewline": "before",
  "aliasAs": "always",
  "tabulateAlias": false,
  "commaPosition": "after",
  "newlineBeforeOpenParen": true,
  "newlineBeforeCloseParen": true,
  "expressionWidth": 50,
  "linesBetweenQueries": 2,
  "denseOperators": false,
  "newlineBeforeSemicolon": false
}
```

**Actual Output**

At the first format, it generates the query (notice the `(1)` on newlines with no spaces before them):
```sql
SELECT CAST(
    SPLIT(key1.path, ',')[
      OFFSET
(1)
    ] AS INT64
  ) AS col1,
  CAST(
    SPLIT(key2.path, ',')[
      OFFSET
(1)
    ] AS INT64
  ) AS col2,
  key3_looooooooooooooong_name.id AS col3
FROM `my_dataset.my_table`;
```

And on formatting again, it adds spaces before the `(1)`:
```sql
SELECT CAST(
    SPLIT(key1.path, ',')[
      OFFSET
        (1)
    ] AS INT64
  ) AS col1,
  CAST(
    SPLIT(key2.path, ',')[
      OFFSET
        (1)
    ] AS INT64
  ) AS col2,
  key3_looooooooooooooong_name.id AS col3
FROM `my_dataset.my_table`;
```

**Small changes to the query**

If I just make a little change in the query to move `key3_looooooooooooooong_name.id AS col3` at the first place in the select statement. It would generate a different result.

```sql
SELECT key3_looooooooooooooong_name.id AS col3,
    CAST(SPLIT(
        key1.path, ',')[OFFSET(1)] AS INT64) AS col1,
    CAST(SPLIT(key2.path, ',')[OFFSET(1)] AS INT64) AS col2
FROM `my_dataset.my_table`;
```

The output from this query. Notice the first offset index `(1)` on newline, and the second offset index in the same line.
```sql
SELECT
  key3_looooooooooooooong_name.id AS col3,
  CAST(
    SPLIT(key1.path, ',')[
      OFFSET
(1)
    ] AS INT64
  ) AS col1,
  CAST(
    SPLIT(key2.path, ',')[
      OFFSET(1)
    ] AS INT64
  ) AS col2
FROM `my_dataset.my_table`;
```

And on formatting again, it would produce:
```sql
SELECT
  key3_looooooooooooooong_name.id AS col3,
  CAST(
    SPLIT(key1.path, ',')[
      OFFSET
        (1)
    ] AS INT64
  ) AS col1,
  CAST(
    SPLIT(key2.path, ',')[
      OFFSET(1)
    ] AS INT64
  ) AS col2
FROM `my_dataset.my_table`;
```

**Expected Output**
I would ideally like it to produce a consistent output in the following form (keep the offset index on the same line):

```sql
SELECT
  key3_looooooooooooooong_name.id AS col3,
  CAST(
    SPLIT(key1.path, ',')[
      OFFSET(1)
    ] AS INT64
  ) AS col1,
  CAST(
    SPLIT(key2.path, ',')[
      OFFSET(1)
    ] AS INT64
  ) AS col2
FROM `my_dataset.my_table`;

```

**Usage**

SQL formatter version: 7.0.1

SQL language: only used it for bigquery

Installed sql-formatter using yarn, and formatting the files like:

```
yarn sql-formatter -o hello.sql -c config.json hello.sql
```

**Comment by nene** [comment]
Thanks for the detailed bug report.

The problem with `OFFSET` is that SQL Formatter isn't aware of the [BigQuery offset operator.](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#array_subscript_operator) Instead it treats it as an `OFFSET` clause in SELECT, and formats it accordingly, as in:

```sql
SELECT
  name,
  price
FROM
  inventory
LIMIT
  10
OFFSET
  125;
```

Fixing it is currently going to require some special fix for the handling of `OFFSET` in BigQuery.

As for the other issues:

- Reformatting of the SQL producing a different result is a known issue #140 (unfortunately no quick fix for this, more of a systematic problem).
- Don't know yet what's causing the different result when reordering SELECT columns. Will investigate. Interestingly it works fine with the default `multilineLists: "always"` option.

**Comment by nene** [comment]
Fixed the issue with use of `OFFSET` in BigQuery.

**Comment by nene** [comment]
Released the `OFFSET` fix in 7.0.2.

**Comment by m-ahmadkhan** [comment]
Awesome! Thanks for the quick fix. I'll give it a try again.
